backend/controllers/taskController.js

```diff
@@ -1,184 +1,167 @@
-});
-
-const getTasksByDay = asyncHandler(async (req, res) => {
-    const date = req.params.day;
-    const userId = req.id;
-
-    let startDate = 0;
-
-    if (date.length === 8) {
-        // parse
-        const year = parseInt(date.substring(0, 4));
-        const month = parseInt(date.substring(4, 6));
-        const day = parseInt(date.substring(6, 8));
-
-        try {
-            startDate = new Date(year, month - 1, day);
-            endDate = new Date(year, month - 1, day + 2);
-        } catch (error) {
-            throw new Error("Invalid Date Input");
-        }
+  const startDateParam = req.query.start;
+  const endDateParam = req.query.end;
+  const userId = req.id;
+
+  let startDate = 0;
+  let endDate = 0;
+
+  if (startDateParam && endDateParam) {
+    if (startDateParam.length === 8 && endDateParam.length === 8) {
+      // parse
+      const year = parseInt(startDateParam.substring(0, 4));
+      const month = parseInt(startDateParam.substring(4, 6));
+      const day = parseInt(startDateParam.substring(6, 8));
+
+      const year_end = parseInt(endDateParam.substring(0, 4));
+      const month_end = parseInt(endDateParam.substring(4, 6));
+      const day_end = parseInt(endDateParam.substring(6, 8));
+
+      try {
+        startDate = new Date(year, month - 1, day);
+        endDate = new Date(year_end, month_end - 1, day_end + 1);
+      } catch (error) {
+        throw new Error("Invalid Date Input");
+      }
     }
-
-    // Check if its a valid date object
     const tasks = await Task.find({
-        user_id: userId,
-        startDate: {
+      $or: [
+        {
+          user_id: userId,
+          startDate: {
             $lte: startDate,
+          },
+          endDate: {
+            $gte: startDate,
+          },
         },
-        endDate: {
+        {
+          user_id: userId,
+          startDate: {
+            $lte: startDate,
+          },
+          endDate: {
             $gte: startDate,
-        }
+            $lte: endDate,
+          },
+        },
+        {
+          user_id: userId,
+          startDate: {
+            $gte: startDate,
+            $lt: endDate,
+          },
+          endDate: {
+            $gt: startDate,
+          },
+        },
+      ],
     })
-        .then((docs) => {
-            res.status(200).json(docs);
-        })
-        .catch((err) => {
-            res.status(500);
-            throw new Error(`Could not fetch doc ${err}`);
-        });
+      .then((docs) => {
+        res.status(200).json(docs);
+      })
+      .catch((err) => {
+        res.status(500);
+        throw new Error(`Could not fetch doc ${err}`);
+      });
     res.status(200).json(tasks);
+  } else if (!startDateParam && !endDateParam) {
+    await Task.find({ user_id: userId })
+      .then((docs) => {
+        res.status(200).json(docs);
+      })
+      .catch((err) => {
+        res.status(500);
+        throw new Error(`Could not fetch doc ${err}`);
+      });
+  } else {
```

Comment:
The else from this portion can be moved to my earlier comment about the `if statement`

Reply:
Actually, this can be left here. This is fine